File: src/main/java/com/example/demo/service/LessonService.java
package com.example.demo.service;

import com.example.demo.rest.dto.lesson.AddLessonRequest;
import com.example.demo.rest.dto.lesson.LessonDepartment;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface LessonService {
    ResponseEntity<List<LessonDepartment>> getSchedule();

    ResponseEntity<LessonDepartment> getSchedule(Long departmentId);

    LessonDepartment addLesson(AddLessonRequest request);

    void delete(Long id);
}

File: src/main/frontend/src/service/api/LessonService.js
import axios from "axios";
import {LESSON_API_URL} from "../constants";

export default class LessonService {
    static async getSchedule() {
        return axios.get(`${LESSON_API_URL}/schedule`)
    }

    static async getDepartmentSchedule(departmentId) {
        return axios.get(`${LESSON_API_URL}/schedule/${departmentId}`)
    }

    static async addLesson(addLessonRequest) {
        return axios.post(LESSON_API_URL, addLessonRequest)
    }
}